CIFAR/resnet.swift

```diff
@@ -0,0 +1,101 @@
+import TensorFlow
+
+/// Original Paper:
+/// "Deep Residual Learning for Image Recognition"
+/// Kaiming He, Xiangyu Zhang, Shaoqing Ren, Jian Sun
+/// https://arxiv.org/abs/1512.03385
+/// using shortcut layer to connect BasicBlock layers (aka Option (B))
+/// see https://github.com/akamaster/pytorch_resnet_cifar10 for explanation
+
+struct Conv2DBatchNorm: Layer {
+    var conv: Conv2D<Float>
+    var norm: BatchNorm<Float>
+
+    public init(
+        filterShape: (Int, Int, Int, Int),
+        strides: (Int, Int) = (1, 1),
+        padding: Padding = .valid
+    ) {
+        self.conv = Conv2D(filterShape: filterShape, strides: strides, padding: padding)
+        self.norm = BatchNorm(featureCount: filterShape.3)
+    }
+
+    @differentiable
+    public func applied(to input: Tensor<Float>, in context: Context) -> Tensor<Float> {
+        var tmp = input
+        tmp = conv.applied(to: tmp, in: context)
+        tmp = norm.applied(to: tmp, in: context)
+        return tmp
+    }
+}
+
+struct BasicBlock: Layer {
+    var layer1: Conv2DBatchNorm
+    var layer2: Conv2DBatchNorm
+    var shortcut: Conv2DBatchNorm
+
+    public init(
```

Comment:
The type is internal. Remove `public`.
```suggestion
   init(
```

Reply:
removed all public calls